### DISCUSSION TITLE
nuxt instance unavailable

### DISCUSSION THREAD

**Discussion by melokki** [nuxt/nuxt#18319] [Questions] [answered]
Hello there
I am using `nuxt3` in a new project I have this question about `useFetch()`

I am validating a form with `vee-validate` and I am trying to send the form values to an api, but I get the folllowing error `Uncaught (in promise) Error: nuxt instance unavailable`

My code looks something like this

```js
const onSubmit = handleSubmit(async (values) => {
   const { data: response } = await useFetch(
    `http://localhost:3333/documents`,
    {
      method: "POST",
      body: {
        data: values,
      },
    }
  )
  console.log(response)
})
```

Is this the correct way and it's something wrong with nuxt, or is it incorrect? If it's incorrect, how should I do it? :thinking: 

Any thoughts?

**Comment by danielroe** [comment, marked as answer]
`useFetch` is a composable meant to be used at the top level, not called in a function. Instead, just use `$fetch`

```js
const onSubmit = handleSubmit(async (values) => {
   const response = await $fetch(
    `http://localhost:3333/documents`,
    {
      method: "POST",
      body: {
        data: values,
      },
    }
  )
  console.log(response)
})
```

**Reply by melokki** [reply]
Aaa, ok, I understand :blush:
Thank you very much , it worked like a charm :pray:
